staging/src/k8s.io/apiserver/pkg/apis/apiserver/types.go

```diff
@@ -48,3 +48,52 @@ type AdmissionPluginConfiguration struct {
 	// +optional
 	Configuration *runtime.Unknown
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConnectivityServiceConfiguration provides versioned configuration for network proxy clients.
+type ConnectivityServiceConfiguration struct {
+	metav1.TypeMeta
+
+	// ConnectionServices contains a list of network proxy client configurations
+	ConnectionServices []ConnectionService
+}
+
+// ConnectionService provides the configuration for a single network proxy client.
+type ConnectionService struct {
+	// Name is the name of the connectivity service.
```

Comment:
s/connectivity service/egress/
?

"connectivity service" is close enough to konnectivity-server to be confusing :/

Reply:
Our names tend to be overloaded. I think we are better of "confusing" connectivity service which is the lookup mechanism (and configuration for the same) for the konnectivity-server, than with a network policy type (egress). At least your in the right space/mechanism with that confusion.

Reply:
Switched to EgressSelector as agreed.